From 7407adf97fe10485a7afce561eb425d26f210ca8 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Thu, 17 Jul 2014 21:27:02 +1000
Subject: [PATCH 10/16] metadata: create Set/GetIsCollection

Will be used to propagate the information that the metadata came from a collection rather than a given season/episode.
That information is only necessary during the lifetime of the Metadatalookup object, so no need to change other classes
---
 mythtv/libs/libmythmetadata/metadatacommon.cpp |    6 +++++-
 mythtv/libs/libmythmetadata/metadatacommon.h   |    3 +++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythmetadata/metadatacommon.cpp b/mythtv/libs/libmythmetadata/metadatacommon.cpp
index b734763..2d0a8ae 100644
--- a/mythtv/libs/libmythmetadata/metadatacommon.cpp
+++ b/mythtv/libs/libmythmetadata/metadatacommon.cpp
@@ -75,6 +75,7 @@ MetadataLookup::MetadataLookup(void) :
     m_collectionref(),
     m_tmsref(),
     m_imdb(),
+    m_iscollection(false),
     m_people(),
     m_studios(),
     m_homepage(),
@@ -210,6 +211,7 @@ MetadataLookup::MetadataLookup(
     m_collectionref(collectionref),
     m_tmsref(tmsref),
     m_imdb(imdb),
+    m_iscollection(false),
     m_people(people),
     m_studios(studios),
     m_homepage(homepage),
@@ -303,7 +305,8 @@ MetadataLookup::MetadataLookup(
     m_releasedate(releasedate),
     m_lastupdated(lastupdated),
     m_runtime(runtime),
-    m_runtimesecs(runtimesecs)
+    m_runtimesecs(runtimesecs),
+    m_iscollection(false)
 {
     m_tracknum = 0;
     m_popularity = 0;
@@ -379,6 +382,7 @@ MetadataLookup::MetadataLookup(
     m_runtime(runtime),
     m_runtimesecs(runtimesecs),
     m_inetref(inetref),
+    m_iscollection(false),
     m_people(people),
     m_trailerURL(trailerURL),
     m_artwork(artwork),
diff --git a/mythtv/libs/libmythmetadata/metadatacommon.h b/mythtv/libs/libmythmetadata/metadatacommon.h
index e77a5c5..a205744 100644
--- a/mythtv/libs/libmythmetadata/metadatacommon.h
+++ b/mythtv/libs/libmythmetadata/metadatacommon.h
@@ -260,6 +260,7 @@ class META_PUBLIC MetadataLookup : public QObject, public ReferenceCounter
     void SetCollectionref(const QString &collectionref)
                              { m_collectionref = collectionref; };
     void SetTMSref(const QString &tmsref) { m_tmsref = tmsref; };
+    void SetIsCollection(bool collection) { m_iscollection = collection; };
     void SetPreferDVDOrdering(bool preferdvdorder)
                              { m_dvdorder = preferdvdorder; };
 
@@ -345,6 +346,7 @@ class META_PUBLIC MetadataLookup : public QObject, public ReferenceCounter
     QString GetCollectionref() const { return m_collectionref; };
     QString GetIMDB() const { return m_imdb; };
     QString GetTMSref() const { return m_tmsref; };
+    bool    GetIsCollection() const { return m_iscollection; };
 
     // People
     QList<PersonInfo> GetPeople(PeopleType type) const;
@@ -430,6 +432,7 @@ class META_PUBLIC MetadataLookup : public QObject, public ReferenceCounter
     QString m_collectionref;
     QString m_tmsref;
     QString m_imdb;
+    bool    m_iscollection;
 
     // People - Video
     const PeopleMap m_people;
-- 
1.7.10.2

